Enumerable-Methods/spec/spec_helper.rb

```diff
@@ -0,0 +1,51 @@
+#spec_helper.rb
+
+require './modules'
+
+describe Enumerable do 
+  describe "#my_map" do 
+    it "iterates through the elements of array" do
+      expect([1,3,5].my_map{|x| x+1}).to match_array [2,4,6]
+    end
+  end
+
+  describe "#my_any" do 
+    it "checks weither any element of an array is complient with a condition" do
+      expect([5,7,8].my_any{|x| x>10}).to eq(false)
+    end
+  end
+
+
+  describe "#my_none" do 
+    it "checks if none of an array's element is complient with a condition" do
+      expect([1,8,"8"].my_none{|x| is_a?(String)}).to be(true)
+    end
+  end
+
+  describe "#my_all" do 
+    it "checks if all the elements are complient with a condition" do
+      expect([1,8,9,2,10].my_all{|x| x>0}).to be(true)
+    end
+
+    it "return an error if the passed array is not flattened" do
+      expect{[1,[8,9],2,10].my_all{|x| x>0}}.to raise_error
```

Comment:
you can use the `raise_error` matcher to specify a block of code raises an error but here there is no specific error given to it

Reply:
```suggestion
      expect{[1,[8,9],2,10].my_all{|x| x>0}}.to raise_error
```
Traceback (most recent call last):
        6: from modules.rb:195:in `<main>'
        5: from modules.rb:45:in `my_all'
        4: from modules.rb:14:in `my_each'
        3: from modules.rb:14:in `each'
        2: from modules.rb:15:in `block in my_each'
        1: from modules.rb:46:in `block in my_all'
modules.rb:195:in `block in <main>': undefined method `>' for [8, 9]:Array (NoMethodError)

using all? on a non flattened array will raise a NoMethodError